test/bdd/agent_steps.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package bdd
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/hyperledger/aries-framework-go/pkg/client/didexchange"
+
+	"github.com/DATA-DOG/godog"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/defaults"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	dbPath = "./db"
+)
+
+var logger = logrus.New()
+
+// AgentSteps
+type AgentSteps struct {
+	bddContext *Context
+}
+
+// NewAgentSteps
+func NewAgentSteps(context *Context) *AgentSteps {
+	return &AgentSteps{bddContext: context}
+}
+
+func (a *AgentSteps) createAgent(agentID, inboundHost, inboundPort string) error {
+	if inboundPort == "random" {
+		inboundPort = strconv.Itoa(mustGetRandomPort(5))
+	}
+	agent, err := aries.New(defaults.WithInboundHTTPAddr(fmt.Sprintf("%s:%s", inboundHost, inboundPort)), defaults.WithStorePath(dbPath+"/"+agentID))
+	if err != nil {
+		return fmt.Errorf("failed to create new agent: %w", err)
+	}
+	ctx, err := agent.Context()
+	if err != nil {
+		return fmt.Errorf("failed to create context: %w", err)
+	}
+	// create new did exchange client
+	didexchangeClient, err := didexchange.New(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create new didexchange client: %w", err)
+	}
+
+	a.bddContext.DIDExchangeClients[agentID] = didexchangeClient
+	// wait until inbound http server start
+	time.Sleep(2 * time.Second)
```

Comment:
please use `listenFor(host string, d time.Duration)` function instead of sleeping

Reply:
can you put the logic of this method ```listenFor(host string, d time.Duration)```

Reply:
```

func listenFor(host string, d time.Duration) error {
	timeout := time.After(d)
	for {
		select {
		case <-timeout:
			return errors.New("timeout: server is not available")
		default:
			conn, err := net.Dial("tcp", host)
			if err != nil {
				continue
			}
			if err := conn.Close(); err != nil {
				return err
			}
			return nil
		}
	}
}

```